CONTRIBUTING.md

```diff
@@ -0,0 +1,127 @@
+* Open a Pull Request from \<user\>/testsuite:\<branch\> → Kuadrant/testsuite:main
+
+**Note for external contributors:** 
+If you are contributing from outside Red Hat, you will need to run the tests on a Kubernetes cluster that does not require internal VPN access. A local KIND cluster (or any accessible Kubernetes cluster) is supported. Tests can be run either locally with Python/Poetry or via the container image. Refer to the [README](https://github.com/Kuadrant/testsuite/blob/main/README.md) for more details on running tests.
+
+### Forking the Repository
+
+Fork the repository first, then clone and navigate into your fork:
+
+```shell
+git clone git@github.com:<your-username>/testsuite.git
+cd testsuite
+```
+
+Add the upstream remote to easily pull the latest changes from the main repository:
+
+```shell
+git remote add upstream git@github.com:Kuadrant/testsuite.git
+```
+
+### Coding Rules / Guidelines
+
+**Test Naming Conventions:**
+
+- Test files use the pattern `test_<feature>.py` (e.g., `test_authorino_metrics.py`)
+- Test functions must use the `test_` prefix and clearly describe the behaviour being tested (e.g., `test_dnspolicy_removal`)
+- Fixture names should be short, lower-case, and reflect their purpose (e.g.,
+gateway, route, authorization)
+- Parametrised tests should use meaningful parameter IDs (ids=) to improve readability in test output
+
+**Common Testing Patterns:**
+
+- Every test and fixture should include a meaningful docstring
+- Mark tests appropriately with pytest markers (`@pytest.mark authorino`, `@pytest.mark.limitador`, etc.)
+- Use `autouse=True` on fixtures when resources should be automatically applied before tests
+- Use `request.addfinalizer()` to ensure cleanup/teardown of resources
+- Follow policy lifecycle: `create → configure → commit → wait_for_ready → delete`
+
+**Test Organization:**
+
+- `singlecluster/` for single-cluster scenarios
+- `multicluster/` for multi-cluster scenarios
+- Group tests by feature area (authorino, limitador, gateway, etc.)
+
+### Reformat, Commit Acceptance & Cleanup
+
+Before committing your changes, make sure the code is properly formatted, and all commit checks pass. Otherwise, your pull request may fail the GitHub Actions **Code Static Analysis** checks.
+
+**Reformat code:**
+
+```shell
+make reformat
+```
+
+This runs ‘black’ through Poetry to automatically format all source files according to the project’s code style.
+
+**Commit acceptance:**
+
+```shell
+make commit-acceptance
+```
+
+This performs a final validation before committing by running:
+
+* **Black** in check mode for code formatting  
+* **Pylint** for linting and code quality  
+* **Mypy** for static type checking
+
+**Cleanup:**  
+Once you’re done testing, if any resources created during the tests remain in the cluster, you can clean them up by running:
+
+```shell
+make clean
+```
+
+### Committing changes & Opening a Pull Request
+
+**1\. Stage all changes:**
+
+```shell
+git add .
```

Comment:
`git add <filename1> <filename2> ...`

`git add .` works perfectly fine usually, there is just a chance that you make a mistake and commit unintended file/secret to the git with it, when you're doing it regularly